app/main.py

```diff
@@ -2,24 +2,91 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(
+            self,
+            min_amount: int,
+            max_amount: int
+    ) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(
+            self,
+            owner: any,
+            name: str
+    ) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(
+            self,
+            instance: any,
+            owner: any
+    ) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(
+            self,
+            instance: any,
+            value: int
+    ) -> None:
+        if not self.min_amount <= value <= self.max_amount:
+            raise ValueError(
+                f"Value should not be less than {self.min_amount}"
+                f"and greater than {self.max_amount}."
```

Comment:
The error message in the ValueError exception is not correct. It says that the value should not be less than min_amount and greater than max_amount, but according to your condition, the value should be in the range between min_amount and max_amount. Please, correct the error message.